lib/AST/Module.cpp

```diff
@@ -1444,7 +1444,8 @@ void SourceFile::getPrecedenceGroups(
 }
 
 void SourceFile::getLocalTypeDecls(SmallVectorImpl<TypeDecl*> &Results) const {
```

Comment:
I might be missing a reference, but I believe the only two uses of this can just be replaced with `getLocalTypeDecls()` - neither use mutates the vector after retrieving them.

Reply:
`Module::getLocalTypeDecl()` calls `FileUnit::getLocalTypeDecls(SmallVectorImpl<TypeDecl*>)`. Since `SerializedASTFile` doesn't hold them as a list of `TypeDecl *` we can't simply make it returning `ArrayRef`.